### PR TITLE
fix bug 1345338 Update titles+metadata on mozilla.org/firefox/*

### PR DESCRIPTION
**Pull request by FrancescoSTL**
## Description
Updating the following pages with [these](https://docs.google.com/spreadsheets/d/1kFpqkkssd-keOyVS13azQS_iJ8ESNtvqjZ5TLy5hBq0/edit#gid=0) meta-decriptions and titles:

- https://www.mozilla.org/en-US/firefox/products/
- https://www.mozilla.org/en-US/firefox/desktop/
- https://www.mozilla.org/en-US/firefox/ios/
- https://www.mozilla.org/en-US/firefox/private-browsing/
- https://www.mozilla.org/en-US/firefox/sync/
- https://www.mozilla.org/en-US/firefox/releases/
- https://www.mozilla.org/en-US/firefox/android/

## Bugzilla link
https://bugzilla.mozilla.org/show_bug.cgi?id=1345338#c12

## Testing
Ensure various meta titles and descriptions are updated as denoted [here](https://docs.google.com/spreadsheets/d/1kFpqkkssd-keOyVS13azQS_iJ8ESNtvqjZ5TLy5hBq0/edit#gid=0).

## Checklist
- [x] Requires l10n changes.

### CODE REVIEW

**Comment by FrancescoSTL** [comment]
Need to change tag to reflect the following:
https://bugzilla.mozilla.org/show_bug.cgi?id=1345338#c14

**Comment by FrancescoSTL** [comment]
Ready for review

**Comment by FrancescoSTL** [comment]
Ready for review

File: bedrock/firefox/templates/firefox/channel/android.html [hunk 1]
```diff
@@ -6,10 +6,30 @@
...
+{% if l10n_has_tag('firefox_title_meta_bug1345338') %}
+  {{ _('Try New Features in a Pre-Release Android Browser | Firefox') }}
+{% else %}
+  {{ _('Download and test future releases of Firefox for desktop, Android and iOS.') }}
+{% endif %}
+{% endblock %}
+
+{% block page_desc %}
+{% if l10n_has_tag('firefox_title_meta_bug1345338') %}
+  {{_('Experience cutting-edge features in a pre-release browser for Android: Firefox Developer Edition, Firefox Beta and Firefox Nightly. Download now!')}}
```
Comment (hoosteeno) [inline bedrock/firefox/templates/firefox/channel/android.html]: With apologies, I think we need to modify this copy -- mobile browsers have different channel names.

> Experience cutting-edge features in a pre-release browser for Android: Firefox Beta, Firefox Aurora and Firefox Nightly. Download now!

File: bedrock/firefox/templates/firefox/channel/android.html [hunk 2]
```diff
@@ -6,10 +6,30 @@
...
+{% else %}
+  {{_('Download and test the latest Firefox for Android features with Aurora, Beta and Nightly builds.')}}
+{% endif %}
+{% endblock %}
 
 {#- This will appear as <meta property="og:description"> which can be used for social share -#}
-{% block page_og_desc %}{{ self.page_desc() }}{% endblock %}
+{% block page_og_desc %}
+{% if l10n_has_tag('firefox_title_meta_bug1345338') %}
+  {{_('Experience cutting-edge features in a pre-release browser for Android: Firefox Developer Edition, Firefox Beta and Firefox Nightly. Download now!')}}
```
Comment (hoosteeno) [inline bedrock/firefox/templates/firefox/channel/android.html]: Same as above.

File: bedrock/firefox/templates/firefox/channel/ios.html [hunk 3]
```diff
@@ -6,10 +6,30 @@
...
+{% if l10n_has_tag('firefox_title_meta_bug1345338') %}
+  {{ _('Try New Features in a Pre-Release iOS Browser | Firefox') }}
+{% else %}
+  {{_('Test beta versions of Firefox for iOS via Apple’s TestFlight program and help make our mobile browser for iPhone, iPad and iPod touch even better.')}}
+{% endif %}
+{% endblock %}
+
+{% block page_desc %}
+{% if l10n_has_tag('firefox_title_meta_bug1345338') %}
+  {{_('Experience cutting-edge features in a pre-release browser for iOS: Firefox Developer Edition, Firefox Beta and Firefox Nightly. Download now!')}}
```
Comment (hoosteeno) [inline bedrock/firefox/templates/firefox/channel/ios.html]: With apologies, I think we need to modify this copy -- mobile browsers have different channel names.

> Experience cutting-edge features in a pre-release browser for iOS via Apple’s TestFlight program. Install now!

File: bedrock/firefox/templates/firefox/channel/ios.html [hunk 4]
```diff
@@ -6,10 +6,30 @@
...
+{% else %}
+  {{_('Test beta versions of Firefox for iOS via Apple’s TestFlight program and help make our mobile browser for iPhone, iPad and iPod touch even better..')}}
+{% endif %}
+{% endblock %}
 
 {#- This will appear as <meta property="og:description"> which can be used for social share -#}
-{% block page_og_desc %}{{ self.page_desc() }}{% endblock %}
+{% block page_og_desc %}
+{% if l10n_has_tag('firefox_title_meta_bug1345338') %}
+  {{_('Experience cutting-edge features in a pre-release browser for iOS: Firefox Developer Edition, Firefox Beta and Firefox Nightly. Download now!')}}
```
Comment (hoosteeno) [inline bedrock/firefox/templates/firefox/channel/ios.html]: Same as above.

File: bedrock/firefox/templates/firefox/android/index.html [hunk 5]
```diff
@@ -9,8 +9,22 @@
...
+{% if l10n_has_tag('firefox_title_meta_bug1345338') %}
+  {{ _('Fast, Free Android Browser for Tablets and Phones | Firefox') }}
+{% else %}
+  {{ _('Firefox for Android — Mobile Web browser — More ways to customize and protect your privacy') }}
+{% endif %}
+{% endblock %}
+
+{% block page_desc %}
+{% if l10n_has_tag('firefox_title_meta_bug1345338') %}
+  {{ _('Join the millions browsing better on tablets and phones. Firefox is fast and private. Download for Android now!') }}
```
Comment (hoosteeno) [inline bedrock/firefox/templates/firefox/android/index.html]: Realizing we need to speak to the mobile install experience:

> Join the millions browsing better on tablets and phones. Firefox is fast and private. Install Firefox for Android now!

File: bedrock/firefox/templates/firefox/ios.html [hunk 6]
```diff
@@ -9,8 +9,21 @@
...
+{% if l10n_has_tag('firefox_title_meta_bug1345338') %}
+  {{ _('Free Mobile Browser App for iPhone, iPad, iOS | Firefox') }}
+{% else %}
+  {{ _('Mobile Web browser for your iPhone, iPad and iPod touch') }}
+{% endif %}
+{% endblock %}
+
+{% block page_desc %}
+{% if l10n_has_tag('firefox_title_meta_bug1345338') %}
+  {{ _('Join the millions browsing better on iPhone and iPad. Firefox is fast and private. Download for iOS now!') }}
```
Comment (hoosteeno) [inline bedrock/firefox/templates/firefox/ios.html]: Oops, mobile install experience isn't "download".

> Join the millions browsing better on iPhone and iPad. Firefox is fast and private. Install Firefox for iOS now!

File: bedrock/firefox/templates/firefox/channel/android.html [hunk 7]
```diff
@@ -6,10 +6,30 @@
...
+{% else %}
+  {{ self.page_desc() }}
```
Comment (jpetto) [inline bedrock/firefox/templates/firefox/channel/android.html]: You don't need the `if/else` block here. `self.page_desc()` will simply repeat the result of the conditional within the `page_desc` block.

File: bedrock/firefox/templates/firefox/channel/desktop.html [hunk 8]
```diff
@@ -6,10 +6,30 @@
...
+{% endblock %}
 
 {#- This will appear as <meta property="og:description"> which can be used for social share -#}
-{% block page_og_desc %}{{ self.page_desc() }}{% endblock %}
+{% block page_og_desc %}
+{% if l10n_has_tag('firefox_title_meta_bug1345338') %}
+  {{_('Experience cutting-edge browser features in pre-release versions: Firefox Developer Edition, Firefox Beta and Firefox Nightly. Download now!')}}
+{% else %}
+  {{ self.page_desc() }}
+{% endif %}
```
Comment (jpetto) [inline bedrock/firefox/templates/firefox/channel/desktop.html]: Same here - can just use `self.page_desc()` without the `if/else` to inherit the value of `page_title` block.

File: bedrock/firefox/templates/firefox/channel/ios.html [hunk 9]
```diff
@@ -6,10 +6,30 @@
...
+{% else %}
+  {{ self.page_desc() }}
+{% endif %}
```
Comment (jpetto) [inline bedrock/firefox/templates/firefox/channel/ios.html]: Just use `self.page_desc()` without the `if/else`.

File: bedrock/firefox/templates/firefox/family/index.html [hunk 10]
```diff
@@ -6,15 +6,24 @@
...
 {% block page_title %}
-{% if l10n_has_tag('fx49_remove_hello') %}
+{% if l10n_has_tag('firefox_title_meta_bug1345338') %}
+  {{ _('Free Web Browser for Android, iOS, Desktop | Firefox') }}
+{% elif l10n_has_tag('fx49_remove_hello') %}
   {{ _('Firefox – Desktop browser, Android, iOS, Sync, Private Browsing') }}
 {% elif l10n_has_tag('tracking_protection') %}
   {{ _('Firefox – Desktop browser, Android, iOS, OS, Hello, Sync, Private Browsing') }}
 {% else %}
   {{ _('Firefox — Desktop browser, Android, OS, Hello, Sync, Marketplace') }}
```
Comment (jpetto) [inline bedrock/firefox/templates/firefox/family/index.html]: Not specifically part of this change, but can you check with Peying to see if we can remove these two fallbacks that mention Hello? These shouldn't be used anymore (for obvious reasons 😁).

File: bedrock/firefox/templates/firefox/channel/desktop.html (hunk 8 above)
Comment (FrancescoSTL) [inline bedrock/firefox/templates/firefox/channel/desktop.html]: I assume you mean the `page_desc` block here, correct @jpetto?

File: bedrock/firefox/templates/firefox/family/index.html (hunk 10 above)
Comment (FrancescoSTL) [inline bedrock/firefox/templates/firefox/family/index.html]: Will do.

File: bedrock/firefox/templates/firefox/channel/desktop.html (hunk 8 above)
Comment (jpetto) [inline bedrock/firefox/templates/firefox/channel/desktop.html]: Ah, yes, sorry. `self.page_desc()` will inherit the value of the `page_desc` block, whatever that turns out to be. 😁

**Comment by FrancescoSTL** [comment]
Updated based upon conversation above. r? @hoosteeno @jpetto

File: bedrock/firefox/templates/firefox/channel/android.html [hunk 11]
```diff
@@ -6,10 +6,24 @@
...
+{% if l10n_has_tag('firefox_title_meta_bug1345338') %}
+  {{_('Experience cutting-edge features in a pre-release browser for Android: Firefox Beta, Firefox Aurora and Firefox Nightly. Install now!')}}
+{% else %}
+  {{_('Download and test the latest Firefox for Android features with Aurora, Beta and Nightly builds.')}}
+{% endif %}
+{% endblock %}
 
 {#- This will appear as <meta property="og:description"> which can be used for social share -#}
-{% block page_og_desc %}{{ self.page_desc() }}{% endblock %}
+{% block page_og_desc %}{ self.page_desc() }}{% endblock %}
```
Comment (jpetto) [inline bedrock/firefox/templates/firefox/channel/android.html]: Accidental deletion - missing one opening `{`.

File: bedrock/firefox/templates/firefox/channel/ios.html [hunk 12]
```diff
@@ -6,7 +6,21 @@
...
+{% else %}
+  {{_('Test beta versions of Firefox for iOS via Apple’s TestFlight program and help make our mobile browser for iPhone, iPad and iPod touch even better.')}}
+{% endif %}
+{% endblock %}
+
+{% block page_desc %}
+{% if l10n_has_tag('firefox_title_meta_bug1345338') %}
+  {{_('Experience cutting-edge features in a pre-release browser for iOS via Apple’s TestFlight program. Install now!')}}
+{% else %}
+  {{_('Test beta versions of Firefox for iOS via Apple’s TestFlight program and help make our mobile browser for iPhone, iPad and iPod touch even better..')}}
```
Comment (jpetto) [inline bedrock/firefox/templates/firefox/channel/ios.html]: Typo - two periods at the end of this string.

**Comment by FrancescoSTL** [comment]
Done @jpetto 🚀

File: bedrock/firefox/templates/firefox/channel/ios.html [hunk 13]
```diff
@@ -6,7 +6,21 @@
 
 {% add_lang_files "firefox/channel/index" %}
 
-{% block page_desc %}{{_('Test beta versions of Firefox for iOS via Apple’s TestFlight program and help make our mobile browser for iPhone, iPad and iPod touch even better.')}}{% endblock %}
+{% block page_title %}
+{% if l10n_has_tag('firefox_title_meta_bug1345338') %}
+  {{ _('Try New Features in a Pre-Release iOS Browser | Firefox') }}
+{% else %}
+  {{_('Test beta versions of Firefox for iOS via Apple’s TestFlight program and help make our mobile browser for iPhone, iPad and iPod touch even better.')}}
```
Comment (peiying2) [inline bedrock/firefox/templates/firefox/channel/ios.html]: The title and the description contain the same text: Test beta versions of Firefox for iOS ...

**Pull request merged**